src/generative_models/encoders.jl

```diff
@@ -0,0 +1,54 @@
+
+"""
+Encoder
+
+Constructs encoder part of VAE: a simple Flux neural network with one hidden layer and two linear output layers for the first two moments of the latent distribution.
+"""
+struct Encoder
+    linear::Any
+    μ::Any
+    logσ::Any
+end
+Flux.@functor Encoder
+
+function Encoder(input_dim::Int, latent_dim::Int, hidden_dim::Int; activation=sigmoid)
+    return Encoder(
+        Dense(input_dim, hidden_dim, activation),       # linear
+        Dense(hidden_dim, latent_dim),                  # μ
+        Dense(hidden_dim, latent_dim),                  # logσ
```

Comment:
Noting just in case that these should also be changed to Flux.Dense

Reply:
Yep they've been changed, I was a bit cofnused with the flux/flux.losses thing